Src/DependencyCollector/Shared/Implementation/ProfilerHttpProcessing.cs

```diff
@@ -385,13 +391,17 @@ private void OnEnd(object exception, object thisObj, object returnValue)
 
                             if (responseObj.Headers != null)
                             {
-                                var targetIkeyHash = responseObj.Headers[RequestResponseHeaders.TargetInstrumentationKeyHeader];
-                                
-                                // We only add the cross component correlation key if the key does not remain the current component.
-                                if (!string.IsNullOrEmpty(targetIkeyHash) && targetIkeyHash != InstrumentationKeyHashLookupHelper.GetInstrumentationKeyHash(telemetry.Context.InstrumentationKey))
+                                var targetAppId = responseObj.Headers[RequestResponseHeaders.TargetAppIdHeader];
+
+                                string myAppId;
+                                if (CorrelationIdLookupHelper.TryGetXComponentCorrelationId(telemetry.Context.InstrumentationKey, this.appIdEndpoint, out myAppId))
                                 {
-                                    telemetry.Type = RemoteDependencyConstants.AI;
-                                    telemetry.Target += " | " + targetIkeyHash;
+                                    // We only add the cross component correlation key if the key does not remain the current component.
+                                    if (!string.IsNullOrEmpty(targetAppId) && targetAppId != myAppId)
+                                    {
+                                        telemetry.Type = RemoteDependencyConstants.AI;
+                                        telemetry.Target += " | " + targetAppId;
+                                    }
                                 }
                             }
                         }
```

Comment:
Should the try catch this is wrapped in handle more than ObjectDisposedException? There's an outer catch over everything but it's not clear to me that it is appropriate to jump out of this logic and skip setting resultcode & success as well as not run EndTracking on exceptions from correlation id.